src/coins.cpp

```diff
@@ -262,21 +214,14 @@ unsigned int CCoinsViewCache::GetCacheSize() const {
     return cacheCoins.size();
 }
 
-const CTxOut &CCoinsViewCache::GetOutputFor(const CTxIn& input) const
-{
-    const CCoins* coins = AccessCoins(input.prevout.hash);
-    assert(coins && coins->IsAvailable(input.prevout.n));
-    return coins->vout[input.prevout.n];
-}
-
 CAmount CCoinsViewCache::GetValueIn(const CTransaction& tx) const
 {
     if (tx.IsCoinBase())
         return 0;
 
     CAmount nResult = 0;
     for (unsigned int i = 0; i < tx.vin.size(); i++)
-        nResult += GetOutputFor(tx.vin[i]).nValue;
+        nResult += AccessCoin(tx.vin[i].prevout).out.nValue;
```

Comment:
I was wondering what happens in this function if an input doesn't exist -- in the old code, it appears we'd `assert()` in `GetOutputFor()`, whereas in the new code we'll just silently return an incorrect value (because nValue is -1 in that case).  Do you think it's worth asserting here that AccessCoin is returning something valid?